### PR TITLE
feat(webapp):improve on scopes input fields and download flow

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes
I added a timestamp in front of the zip file name to produce `1646812165-nango-integrations.zip`, ensuring each download file has a unique name. Additionally, I included functionality to disable the download spinner by invoking `setIsDownloading(false) `whenever we encounter an error. I also implemented functionality to add scopes whenever the user clicks on  `Add new scope`.
scopes

### CODE REVIEW

File: packages/shared/lib/services/file/local.service.ts [hunk 1]
```diff
@@ -184,8 +184,10 @@ class LocalFileService {
             return;
         });
 
+        const timestamp = Math.floor(new Date().getTime() / 1000).toString();
+        const filename = `${timestamp}-nango-integrations.zip`;
         res.setHeader('Content-Type', 'application/zip');
-        res.setHeader('Content-Disposition', `attachment; filename=nango-integrations.zip`);
+        res.setHeader('Content-Disposition', `attachment; filename=${filename}`);
```
Comment (khaliqgant) [inline packages/shared/lib/services/file/local.service.ts]: Did this work for you? I downloaded a file and the name was still `nango-integrations.zip`. Editing the name in the [FlowPage](https://github.com/NangoHQ/nango/blob/master/packages/webapp/src/pages/Integration/FlowPage.tsx#L93) worked as expected h/e

**Review by khaliqgant — CHANGES_REQUESTED** [review]
See note about the download name -- relates to the change in `file/local.service.ts` and `file/remote.service.ts`

File: packages/webapp/src/pages/Integration/FlowPage.tsx [hunk 2]
```diff
@@ -89,8 +90,9 @@ export default function FlowPage(props: FlowPageProps) {
         const blob = await response.blob();
         const url = window.URL.createObjectURL(blob);
         const link = document.createElement('a');
+        const timestamp = Math.floor(new Date().getTime() / 1000).toString();
         link.href = url;
-        link.download = 'nango-integrations.zip';
+        link.download = timestamp + '-nango-integrations.zip';
```
Comment (TBonnin) [inline packages/webapp/src/pages/Integration/FlowPage.tsx]: I would go for `nango-integration-TIMESTAMP.zip` but I am nitpicking

**Review by TBonnin — APPROVED** [review]

File: packages/webapp/src/pages/Integration/FlowPage.tsx (hunk 2 above)
Comment (khaliqgant) [reply packages/webapp/src/pages/Integration/FlowPage.tsx]: Agreed!

Comment (hassan254-prog) [reply packages/webapp/src/pages/Integration/FlowPage.tsx]: Hi @khaliqgant , just updated the pull request with the notes above .

**Pull request merged**